### PR TITLE
Update SDK documentation

### PR DESCRIPTION
**Pull request by JenkoB**

### CODE REVIEW

File: docs/intro.rst [hunk 1]
```diff
@@ -124,66 +124,193 @@ In addition to samples and their data, collections may contain
...
+    #save the annotation
+    reads.save()
+
+We can also define descriptor and descriptor schema directly when calling
+'res.run' function as described in The run method section.
+
+Managing presamples and samples
+===============================
+
+When you upload new data that belongs to the biological sample or run analysis
```
Comment (jkokosar) [inline docs/intro.rst]: When a new data object that represents a biological sample (i.e. fastq files, bam files) is uploaded to the database, the unannotated sample (presample) is automatically created. When a data object that belongs to an existing (pre)sample is used as an input to trigger new analysis, the output of the new analysis is automatically attached to an existing (pre)sample.

File: docs/intro.rst [hunk 2]
```diff
@@ -124,66 +124,193 @@ In addition to samples and their data, collections may contain
...
+on such data, process automatically create new (pre)sample for created object
+or add it to the an existing sample. For example if we upload fastq reads file
+and annotate presample to become sample, aligned bam and bai files will be also
+part of this same sample.
+
+When new sample is created, it is called ``presample``, which means that it is
```
Comment (jkokosar) [inline docs/intro.rst]: Unannotated samples (presamples) have a ``presample``property set to ``True``. To annotate a presample, fill its descriptor ...

File: docs/intro.rst [hunk 3]
```diff
@@ -124,66 +124,193 @@ In addition to samples and their data, collections may contain
...
+    reads.sample.name = 'My sample 2'
+    reads.sample.slug = 'my-sample-2'
+    reads.sample.save()
+
+
+Managing collections
+====================
+
+To keep a clear structure of data objects and samples we can create collections.
+Here is an example how to create new collection and add/remove data to/from
```
Comment (jkokosar) [inline docs/intro.rst]: Here is an example of how to..

File: docs/intro.rst [hunk 4]
```diff
@@ -124,66 +124,193 @@ In addition to samples and their data, collections may contain
...
+    #define the collection slug
+    test_collection.slug = 'test-collection'
+
+    #save new collection
+    test_collection.save()
+
+    #add data to collection
+    test_collection.add_data(reads)
+
+    #remove data to collection
```
Comment (jkokosar) [inline docs/intro.rst]: Add also an example of how to add (and remove) samples to/from a collection.

File: docs/run.rst [hunk 5]
```diff
@@ -168,6 +168,21 @@ What just happened?
 * Data object will be included in collections 1 and 2.
 * Data object name was set to "my_favourite_name".
 
+Get or run
+----------
+
+Get or run method, checks if the the process have been already run with the
+same inputs. If it was, it directly return the outputs, otherwise it runs the
```
Comment (jkokosar) [inline docs/run.rst]: returns

File: docs/tutorial.rst [hunk 6]
```diff
@@ -168,6 +230,9 @@ then download the coverage track.
    # Download the coverage track
    bigwig.download()
 
+Writing a new process
```
Comment (jkokosar) [inline docs/tutorial.rst]: Requires a change in the .yml syntax

Comment (jkokosar) [inline docs/tutorial.rst]: `- slug: max-reads            # User readable unique identifier
  name: Max reads            # Process name
  requirements:
    expression-engine: jinja
  data_name: Chromosome      # The name of data objects created by the process
  version: 1.0.0             # Process version
  type: data:bam:stats       # The type of the process and data objects created by it
  input:
    - name: bam
      label: BAM file
      type: data:alignment:bam
  output:
    - name: stats
      label: Stats
      type: basic:file
    - name: chr
      label: Chr (max reads)
      type: basic:string
  run:                       # The algorithm in bash using Django template tags
    runtime: polyglot
    language: bash
    program: |
      # Compute sequence alignment statistics
      samtools idxstats '{{bam.bam.file}}' > stats.txt

      # Save the statistics file
      re-save-file stats stats.txt

      # Find chromosome with the largest number of aligned reads
      CHR=`<my_script>.py stats.txt`

      # Save the chromosome name
      re-save chr "${CHR}"`

**Comment by mstajdohar** [comment]
@JenkoB, please address the comments.

**Comment by JenkoB** [comment]
I have in my to do list :) I will finish till the end of the week.

**Comment by jkokosar** [comment]
@mstajdohar Please merge and publish to PyPi

**Pull request merged**